python/mosaic/utils/kepler_magic.py

```diff
@@ -101,8 +101,25 @@ def mosaic_kepler(self, *args):
                 data = data.withColumn(
                     feature_name, lower(conv(col(feature_name), 10, 16))
                 )
+        if feature_type == "bng":
+            data = (data
+                .withColumn(feature_name, grid_boundaryaswkb(feature_name))
+                .withColumn(feature_name, st_geomfromwkb(feature_name))
+                .withColumn(
+                    feature_name,
+                    st_transform(st_setsrid(feature_name, lit(27700)), lit(4326))
+                )
+                .withColumn(feature_name, st_aswkt(feature_name)))
         elif feature_type == "geometry":
             data = data.withColumn(feature_name, st_astext(col(feature_name)))
+        elif feature_type == "bng_geometry":
```

Comment:
1) There is no documentation for the `bng_geometry`.
2) Technically this is not a "bng_geometry" (It is not linked to the BNG grid system, but rather to a specific CRS EPSG:27700).
I would prefer to add an optional CRS parameter to the "geometry" plotting, to make it valid for any coords without linking it to a specific grid.

Reply:
That can be done.
Not a big change.
I will update now.